test/utils/matchers.go

```diff
@@ -106,7 +107,7 @@ func (matcher *exitCleanlyMatcher) Match(actual interface{}) (success bool, err
 	}
 
 	// Exit status is 0. Now check for anything on stderr
-	if stderr != "" {
+	if stderr != "" && os.Getenv("NO_TEST_CACHE") == "" {
```

Comment:
Yikes. Please add an enormous FIXME comment here explaining why the NO_TEST_CACHE exclusion is necessary and when it can be removed. I'm really really uncomfortable with this, is there any other way to be more selective about when/where warnings are silenced?

Reply:
Many tests fail right now with `NO_TEST_CACHE` because they perform an image pull.

This one was the easiest fix, the alternative would be to filter out any output generated by a pull, or pass `-q` to each command that could cause a pull.

Reply:
I thought I already had added `-q` to any tests that might do a pull. Do you have a log handy, showing sample failures?

Also, in the other diff, please never say "for now". Please always include exact datestamps and, if possible, issue numbers, so a future reader can think "hmmm this should be fixed by now".

Reply:
I've added a "revert" patch to the current PR so we can get a list of the failing tests

Reply:
I was wrong, it seems we don't need this change